docs/contrib/workflow.md

```diff
@@ -0,0 +1,243 @@
+# Contribution to OpenKilda project
+
+Anyone can contribute to the OpenKilda project. This instruction describes how
+OpenKilda contribution looks like: from an idea to OpenKilda release.
+
+# Workflow
+
+## Idea
+
+Before adding a new feature or making an improvement to OpenKilda,
+it is recommended to discuss the idea with other contributors first.
+To find current active contributors, you can view a list of
+[open pull requests](https://github.com/telstra/open-kilda/pulls) and select
+the author of a recent pull request to send a message about your idea.
+They will invite you to a forum for idea discussions.
+
+If you don't have an idea, but you still want to contribute, please take a look on a
+list of [open issues](https://github.com/telstra/open-kilda/issues). If you find
+an open and unassigned issue (of course you will) you can assign it to yourself and
+start to work on it.
+
+### Design
+
+If you are working on a small feature, fix or improvement you can skip the design step.
```

Comment:
I think there is a few points missing here: 
- we need the design doc to be reviewed, challenged, understood in terms of scope, testing, relation to other features, and so on. Before starting with the implementation, it is required get approvals. A person could cross fingers instead of creating a doc for a small change, but they might receive a comment that requests changes and it might require to do much more work comparing that if the same comment received during design doc review.
- when even a small fix makes design documentation incorrect, then either that doc must be corrected accordingly or an issue is created (maybe it is enough just to specify the doc and the issue).
- the life cycle of docs change is pretty similar to other code changes (PRs, labels ready for test and all other stuff)